File: BSSE1430/preprocessing.py/preprocess.py
def padding(message_bin):
    msgLen = len(message_bin)
    mod = msgLen%128
    if(mod<112):
        message_bin+=b'\x80'
        for i in range(112-mod-1):
            message_bin+=b'\x00'
    else:
        message_bin+=b'\x80'
        for i in range(128-mod-1):
            message_bin+=b'\x00'
     
    message_bin+=(msgLen*8).to_bytes(16)   #binary in big-endian format
    return message_bin

    
def main():
    file_read = open("SHAInput.txt", 'r', encoding='utf-8')
    message = file_read.read()
    file_read.close()
    print(message)
    print(len(message))
    
    padded = padding(bytes(message, 'utf-8'))
    print(padded)
    # get_binary(padded.hex())
    print(padded.hex())
    # print(len(padded.hex()))
    
    res = ''.join(format(ord(i), '08b') for i in message)
    print("The string after binary conversion : " + str(res))

if __name__=='__main__':
    main()
    
    


# print("The original string is : " + str(test_str))

# using join() + ord() + format()
# # Converting String to binary
# res = ''.join(format(ord(i), '08b') for i in test_str)

# # printing result 
# print("The string after binary conversion : " + str(res))

File: BSSE1430/preprocessing.py/preprocess.cpp
#include<bits/stdc++.h>
using namespace std;

string padding(string s)
{
    stringstream ss;
    for(int i=0; i<s.size(); i++)
    ss<<bitset<8>(s[i]);
    string padded = ss.str();
    int len = padded.size();
    int add;
    int mod = padded.length()%1024;

    if((1024-mod)>=128)
    add = 1024-mod;
    else if((1024-mod)<128)
    add = 2048 - mod;

    padded+="1";
    for(int i=0; i<add-129; i++)
    padded+="0";

    string L = std::bitset<128>(len).to_string();
    padded+=L;

    return padded;
}

void getHash(string message)
{
    string padded = padding(message);
    cout<<padded<<endl;
    cout<<"Lenth of padded: "<<padded.length();
}

int main()
{
    string message;
    getline(cin, message);
    cout<<"Length: "<<message.length()<<endl;
    cout<<"Binary Length: "<<message.length()*8<<endl;
    
    getHash(message);
    return 0;
}